web_src/js/components/DiffCommitSelector.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div
+    class="ui jump dropdown basic button custom"
+    @click.stop="menuVisible = !menuVisible" @keyup.enter="menuVisible = !menuVisible"
+    :data-tooltip-content="locale.filter_changes_by_commit"
+  >
+    <svg-icon name="octicon-git-commit"/>
+    <div class="menu left transition commit-selector-menu" :class="{visible: menuVisible}" v-if="menuVisible" v-cloak>
+      <a class="vertical item gt-df gt-fc gt-gap-1" :href="issueLink + '/files' + queryParams">
+        <div class="gt-ellipsis">{{ locale.show_all_commits }}</div>
+        <div class="gt-ellipsis text light-2">{{ locale.stats_num_commits }}</div>
+      </a>
+
+      <div class="divider"/>
+      <a v-if="lastReviewCommitSha != null" class="vertical item gt-df gt-fc gt-gap-1" @click="changesSinceLastReviewClick()">
+        <div class="gt-ellipsis">{{ locale.show_changes_since_your_last_review }}</div>
+        <div class="gt-ellipsis text light-2">{{ commitsSinceLastReview }} commits</div>
+      </a>
+
+      <template v-for="commit in commits" :key="commit.id">
+        <div class="divider"/>
+        <div class="vertical item gt-df gt-fr gt-gap-2" :class="{selected: commit.selected}" @click.exact="commitClicked(commit.id)" @click.shift.exact.stop.prevent="commitClickedShift(commit)">
```

Comment:
I really have difficulty to read code with a lot of `gt` patches ....  I can't tell what is `gt-fc` or `gt-fr` every time .....

Reply:
Yes we should refactor to tailwind-compatible classnames eventually, but I don't see a big issue here.